.github/workflows/macos.yml

```
@@ -38,24 +38,8 @@ jobs:
   build_and_test:
     strategy:
       matrix:
-        os: [macos-13, macos-14, macos-15]
-        compiler: [g++-13, g++-14, clang++]
-        include:
-          - compiler: clang++
-            ccompiler: clang
-          - compiler: g++-13
-            ccompiler: gcc-13
-          - compiler: g++-14
-            ccompiler: gcc-14
-        exclude:
-          - os: macos-14
-            compiler: g++-13
-          - os: macos-14
-            compiler: g++-14
-          - os: macos-15
-            compiler: g++-13
-          - os: macos-15
-            compiler: g++-14
+        os: [macos-14, macos-15]
+        compiler: [g++-13, g++-14, g++-15, clang++]
```

Due to compatibility issues, we haven't tested compilation with `g++` on Apple Silicon machines (`macos-14` and `macos-15`) before (see the now redundant `exclude` statements), and I'd recommend keeping it this way, as indicated by the currently running tests, which seem to be stuck.

```suggestion
        compiler: [clang++]
```